backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/IModularContractTypeHandler.java

```diff
@@ -40,6 +46,8 @@
  */
 public interface IModularContractTypeHandler<T>
 {
+	IFlatrateBL flatrateBL = Services.get(IFlatrateBL.class);
```

Comment:
hooking services in a static way is NOK.
As Ruxi suggested, consider moving those methods to service.

Reply:
when moved to service it creates a circular dependency between the handler and service. we can add it inside the function block or send it as a function parameter. wdyt ?